src/ocrd_anybaseocr/cli/ocrd_anybaseocr_layout_analysis.py

```diff
@@ -0,0 +1,227 @@
+
+class OcrdAnybaseocrLayoutAnalyser(Processor):
+    
+    @cached_property
+    def executable(self):
+        return 'ocrd-anybaseocr-layout-analysis'
+
+    def setup(self):
+        if not tf.config.list_physical_devices('GPU'):
+            self.logger.error("Your system has no CUDA installed. No GPU detected.")
+
+        self.last_result = [] 
+        self.logID = 0 # counter for new key
+        self.logIDs = defaultdict(int) # dict to keep track of previous keys for labels other then chapter or section
+        self.log_id = 0 # var to keep the current ongoing key
+        self.log_links = {}
+        self.first = None
+        
+        assert self.parameter
+        if 'model_path' not in self.parameter:
+            self.parameter = {'model_path': str(resource_filename(f'ocrd_anybaseocr.models', 'structure_analysis')), **self.parameter}
+        if 'class_mapping_path' not in self.parameter:
+            self.parameter = {'class_mapping_path': str(resource_filename(f'ocrd_anybaseocr.models', 'mapping_densenet.pickle')), **self.parameter}
+
+        model_path = Path(self.resolve_resource(self.parameter['model_path']))
+        class_mapper_path = Path(self.resolve_resource(self.parameter['class_mapping_path']))
+        if not model_path.exists():
+            self.logger.critical("Layout classfication `model_path` was not found at '%s'", model_path)
+            sys.exit(1)
+        self.logger.info('Loading model from file %s', str(model_path))
+        self.model = self.create_model(str(model_path))
+        # load the mapping
+        pickle_in = open(str(class_mapper_path), "rb")
+        class_indices = pickle.load(pickle_in)
+        self.label_mapping = dict((v,k) for k,v in class_indices.items())
+
+    def process_page_file(self, *input_files : Optional[OcrdFileType]) -> None:
+        self.logger.info("Overridden process_page_file")
+        for input_file in input_files:
+            pcgts = page_from_file(input_file)
+            # No output, just METS modified, so ignore OcrdPageResult
+            _ = self.process_page_pcgts(pcgts, page_id=input_file.pageId)
+
+    def process_page_pcgts(self, *input_pcgts: Optional[OcrdPage], page_id: Optional[str] = None) -> OcrdPageResult:
+        pcgts = input_pcgts[0]
+        assert pcgts
+        result = OcrdPageResult(pcgts)
+        page = pcgts.get_Page()
+        self.logger.info("INPUT FILE %s", page_id)
+        page_image, _, _ = self.workspace.image_from_page(page, page_id, feature_selector='binarized')
+        img_array = pil2array(page_image.resize((500, 600), Image.LANCZOS))
+        img_array = img_array / 255
+        img_array = img_array[np.newaxis, :, :, np.newaxis]            
+        results = self._predict(img_array)
+        self.logger.info(results)
+        #self.workspace.mets.set_physical_page_for_file(input_file.pageId, input_file)
+        self.create_logmap_smlink(pcgts)
+        self.write_to_mets(results, page_id)
+        return result
```

Comment:
This entire approach does not work with the METS Server, which we cannot disallow beforehand, because `max_workers` only applies to the multiprocessing part. So running with METS Server would always fail in that method (raising something about `ClientSideOcrdMets` not provinding access to some attributes).

But we could do better: as examples like ocrd_pagetopdf or ocrd_cis/ocrd_cis/postcorrect/cli.py illustrate, one can do an extra METS serialization step at the end of `process_workspace` to ensure we get access to the METS file directly (and then have the METS Server reload itself, if present)

Reply:
To be more concrete: on each page, you would just record enough information into a processor attribute dict. Then after the page loop (in `process_workspace`, after the super call), you act on that dict: 
- if necessary (ClientSideOcrdMets): 
  - `self.workspace.mets.save_mets()`, 
  - instantiate a new workspace (to get a direct METS)
  - do the METS update
  - save the new workspace METS
  - `self.workspace.mets.reload_mets()`
- otherwise: do the METS update

Reply:
> Not sure whether the final `workspace.mets.reload()` or the `shutdown` are necessary.

Yes, the `reload` is necessary so the (next processor in the) workflow can proceed with the _correct_ (new) version of the METS.

`shutdown` should actually `del self.model` if it exists. The `reset` should happen at the end of `process_workspace`. And `log_id` / `log_links` / `logID` / `logIDs` do not need to be attributes anymore – they could now be passed in to `write_to_mets`. In contrast, `self.page_labels` can be directly used by that function and does not need to be passed in. In fact, that function could itself loop over it.

Reply:
e0e63db I moved the `self.reset()` call and changed `self.shutdown()` accordingly, but I don't want to mess with the attributes/method params right now. Feel free to push if you have the stomach for that refactoring :)

Reply:
ok, I'll test locally and see what I can do

Reply:
https://github.com/OCR-D/ocrd_anybaseocr/pull/113